ion/symboltoken.go

```diff
@@ -97,14 +100,38 @@ func (st *SymbolToken) Equal(o *SymbolToken) bool {
 	return false
 }
 
+func getSystemSymbolMapping(symbolTable SymbolTable, symbolName string) string {
+	// If we have a symbol name of the form '$n' for some integer n,
+	// then we want to use the corresponding system symbol name.
+	if len(symbolName) > 1 && symbolName[0] == '$' {
+		if id, err := strconv.Atoi(symbolName[1:]); err == nil {
+			if systemSymbolName, ok := symbolTable.FindByID(uint64(id)); ok {
+				return systemSymbolName
+			}
+		}
+	}
+
+	return ""
```

Comment:
While it is not the case today, it is possible for a system symbol table to contain the empty string.

Reply:
Addressed.